STM32F405RGTx/Core/Src/rtos_threads/measurements_thread.cpp

```diff
@@ -68,7 +98,11 @@ void MeasurementsThread::runMeasurements(void* args) {
     startADCandDMA();
 
     while (1) {
-        // TODO: need to figure out what needs to be done here, is the function needed?
+        // Process the acceleration data here...
+
+        osThreadFlagsWait(0x00000001U, osFlagsWaitAll, 0U);
```

Comment:
Just an implementatoin question, is the goal to see if the flag is set and if it is, process data, otherwise just go back to acceleration or are you supposed to wait for the flag to be set and then process data before going back to acceleration?

In the second one, the acceleration data measurements will be blocked until the flag is set essentially, so just wondering if this is fine or not.

Reply:
I think it makes most sense to go with the first. There's no point in waiting for the flag to be set before getting acceleration data.